actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,17 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+
+      def extract_variant!
+        if parameters.key?(:format)
+          segments = parameters[:format].to_s.split("+")
```

Comment:
If you can get variant assigned from that, it would be a better way to go than what you now, I think :smile: